integration_tests/src/main/python/parquet_write_test.py

```diff
@@ -262,19 +261,20 @@ def writeParquetUpgradeCatchException(spark, df, data_path, spark_tmp_table_fact
         df.coalesce(1).write.format("parquet").mode('overwrite').option("path", data_path).saveAsTable(spark_tmp_table_factory.get())
     assert e_info.match(r".*SparkUpgradeException.*")
 
-# TODO - https://github.com/NVIDIA/spark-rapids/issues/1130 to handle TIMESTAMP_MILLIS
 # TODO - we are limiting the INT96 values, see https://github.com/rapidsai/cudf/issues/8070
-@pytest.mark.parametrize('ts_write_data_gen', [('INT96', limited_int96()), ('TIMESTAMP_MICROS', TimestampGen(start=datetime(1, 1, 1, tzinfo=timezone.utc), end=datetime(1582, 1, 1, tzinfo=timezone.utc)))])
+@pytest.mark.parametrize('ts_write_data_gen', 
+                        [('INT96', limited_int96()), 
+                         ('TIMESTAMP_MICROS', TimestampGen(start=datetime(1, 1, 1, tzinfo=timezone.utc), end=datetime(1582, 1, 1, tzinfo=timezone.utc))), 
+                         ('TIMESTAMP_MILLIS', TimestampGen(start=datetime(1, 1, 1, tzinfo=timezone.utc), end=datetime(1582, 1, 1, tzinfo=timezone.utc)))])
 @pytest.mark.parametrize('rebase', ["CORRECTED","EXCEPTION"])
 def test_ts_write_fails_datetime_exception(spark_tmp_path, ts_write_data_gen, spark_tmp_table_factory, rebase):
     ts_write, gen = ts_write_data_gen
     data_path = spark_tmp_path + '/PARQUET_DATA'
     int96_rebase = "EXCEPTION" if (ts_write == "INT96") else rebase
-    date_time_rebase = "EXCEPTION" if (ts_write == "TIMESTAMP_MICROS") else rebase
+    date_time_rebase = "EXCEPTION" if (ts_write == "TIMESTAMP_MICROS" or ts_write == "TIMESTAMP_MILLIS") else rebase
```

Comment:
I'm really confused by this change at all.

The test is writing data out that it expects should throw a "SparkUpgradeException" for both the CPU and the GPU or else the the test fails.

So if we are writing out INT96 values we limit the range that is supported for them because we can corrupt things? https://github.com/rapidsai/cudf/issues/8070, but the range shows that it is possible to have values outside of the range that would cause an exception. I guess we are just lucky that we are generating values that would need a rebase.

For TIMESTAMP_MICROS we expect both the CPU and the GPU to throw the "SparkUpgradeException" only if we set `date_time_rebase` to "EXCEPTION"  I would expect this because the CPU will actually correct the out of range values if it is not set to EXCEPTION, but we will not.

For TIMESTAMP_MILLIS I would expect both the CPU and the GPU to behave the same as for MICROS.  The difference is that before the GPU was throwing an assertion error instead of the actual desired exception.  Am I missing something with this?

Reply:
I updated the range of these `TimestampGen`, because timestamps before 1900-01-01T00:00:00Z can be ambiguous. The 1582 limit is for dates. [ref](https://github.com/apache/spark/blob/ce359bcf32bd16d567a67ea717be9d32b3ab0df3/docs/sql-error-conditions-inconsistent-behavior-cross-version-error-class.md?plain=1#L417)

But I'm not sure if I got where you are confused with this change. I want to let it test TIMESTAMP_MILLIS the same way we test TIMESTAMP_MICROS before, because for TIMESTAMP_MILLIS we expect both CPU and GPU to behave the same as for MICROS, as you said. 

This change will not change how we test `ts_write` is INT96, I think.

Reply:
Yes sorry my bad. You added in the TIMESTAMP_MILLS check to make it the same. Not took it away. I guess I just got confused on that part.

Reply:
> The 1582 limit is for dates. ref

So before your change the INT96 end date in the range was wrong since the type of this column is a datetime and you really want to go up to 1900/1/1  to see the rebase issues. That seems like a good change.

I am confused about not going before 1677/9/22 for INT96 (prior to the change it started at 1/1/1). What's the reason for that change?

Reply:
> I am confused about not going before 1677/9/22 for INT96 (prior to the change it started at 1/1/1). What's the reason for that change?

It started at 1677/9/22 prior to the change because of the INT96 overflow issue.

```python
# TODO - we are limiting the INT96 values, see https://github.com/rapidsai/cudf/issues/8070
def limited_int96():
    return TimestampGen(start=datetime(1677, 9, 22, tzinfo=timezone.utc), end=datetime(2262, 4, 11, tzinfo=timezone.utc))
```